beaglebone/gps/src/__main__.py

```diff
@@ -35,38 +80,85 @@ def main():
                 # datalist[5],
                 # ",", datalist[6], "Ground speed(knots):", datalist[7],
                 # "Track angle(in Degrees True):", datalist[8])
-                gps.timeStamp = datalist[1]
-                gps.latitude = datalist[3]
-                gps.latitudeDirection = datalist[4]
-                gps.longitude = datalist[5]
-                gps.longitudeDirection = datalist[6]
-                gps.trackAngle = datalist[8]
+
+                # time
+                if(datalist[1] != ""):
+                    gps.timeStamp = float(datalist[1])
+                else:
+                    gps.timeStamp = tempTimeStamp
+                # dddmm.mmmm format (degrees, minutes, minutes/seconds
+                if(datalist[3] != ""):
+                    if(datalist[4] == "N"):
+                        gps.latitude_deg = int(float(datalist[3]) / 100)
+                        gps.latitude_min = float(datalist[3]) % 100
+                    elif(datalist[4] == "S"):
+                        gps.latitude_deg = int(float(datalist[3]) / 100) * -1.0
+                        gps.latitude_min = float(datalist[3]) % 100
+                else:
+                    gps.latitude_deg = 0
+                    gps.latitude_min = 0
+                if(datalist[5] != ""):
+                    if(datalist[6] == "W"):
```

Comment:
West should be negative rather than east the way you have it. Additionally, both deg and min should include a negative sign in the case of West or South.